devices/debian.py

```diff
@@ -208,7 +208,12 @@ def run_cleanup_cmd(self):
     def sudo_sendline(self, s):
         if self.username != "root":
             s = "sudo " + s
-        return super(DebianBox, self).sendline(s)
+        super(DebianBox, self).sendline(s)
+        idx = self.expect(["password for .*:"] + self.prompt,timeout=20)
+        if idx == 0:
+            self.sendline(self.password)
+            self.expect(self.prompt)
+        return self.before
```

Comment:
Was there a particular reason to return self.before vs. the "number of bytes written" that sendline in the pexpect library would have sent?

I think you should save the return value from 'super(DebianBox, self).sendline(s)' and probably return that instead.

Reply:
The self.before returns the complete output hence I can use it for verification in the places I call this function, whereas if I store the  return value of super(DebianBox, self).sendline(s) and return it, it just gives me bytes.
For Eg: sudo iw wlan1 scan | grep "SSID: " would return the list of SSID which I can fetch from self.before

Reply:
As I commented in your other change, I think we need to just call expect after this function to exact the desired text. I think that means you need to remove the self.expect(self.prompt) so you don't eat that.

sudo_sendline() should not expect anything after the password related messages.